<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialWidget</class>
 <widget class="QWidget" name="SerialWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>685</width>
    <height>495</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QSerialDevice Demo</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="7" column="0">
    <widget class="QPushButton" name="rtsButton">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>20</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Toggle the state of the Request-To-Send line on the port</string>
     </property>
     <property name="text">
      <string>Set RTS</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QLabel" name="lineLabel">
     <property name="text">
      <string>Current State of Serial Lines</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QPushButton" name="dtrButton">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>20</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Toggle the state of the Data-Terminal-Ready line on the port</string>
     </property>
     <property name="text">
      <string>Set DTR</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <layout class="QHBoxLayout" name="horizontalSerialLines">
     <item>
      <widget class="Line" name="line_divider0">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="dsrLabel">
       <property name="text">
        <string>DSR</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider1">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="dtrLabel">
       <property name="text">
        <string>DTR</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="ctsLabel">
       <property name="text">
        <string>CTS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="rtsLabel">
       <property name="text">
        <string>RTS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider4">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="dcdLabel">
       <property name="text">
        <string>DCD</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider5">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="ringLabel">
       <property name="text">
        <string>RING</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider6">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="leLabel">
       <property name="text">
        <string>LE</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_divider7">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0" rowspan="2" colspan="2">
    <layout class="QGridLayout" name="gridMainLayout">
     <item row="0" column="0" colspan="3">
      <widget class="QLabel" name="lblPortConnect">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>15</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>15</height>
        </size>
       </property>
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>1</number>
       </property>
       <property name="midLineWidth">
        <number>0</number>
       </property>
       <property name="text">
        <string>Serial Port Configuration</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="lblAvailablePorts">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>20</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>Available Ports:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="portBox">
         <property name="toolTip">
          <string>List of available ports</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="controlButton">
         <property name="font">
          <font>
           <pointsize>8</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="toolTip">
          <string>Open the currently selected port</string>
         </property>
         <property name="text">
          <string>Open</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="ioButton">
         <property name="text">
          <string>Input/Output</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="infoButton">
         <property name="toolTip">
          <string>Get info on currently selected port</string>
         </property>
         <property name="text">
          <string>Get info</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="3" column="1">
      <spacer name="horizontalSpacerMiddle">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="3" column="2">
      <widget class="QGroupBox" name="groupOptions">
       <property name="minimumSize">
        <size>
         <width>300</width>
         <height>295</height>
        </size>
       </property>
       <property name="title">
        <string> Port Options </string>
       </property>
       <widget class="QWidget" name="gridLayoutWidget_2">
        <property name="geometry">
         <rect>
          <x>9</x>
          <y>19</y>
          <width>282</width>
          <height>281</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayoutPortOptions">
         <item row="0" column="0">
          <widget class="QLabel" name="lblBaudRate">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>20</height>
            </size>
           </property>
           <property name="text">
            <string>Baud rate:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="baudBox">
           <property name="minimumSize">
            <size>
             <width>125</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>List of available baud rates for the current port</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lblDataBits">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Data bits:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="dataBox">
           <property name="minimumSize">
            <size>
             <width>125</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lblParity">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Parity:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="parityBox">
           <property name="minimumSize">
            <size>
             <width>125</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="lblStopBits">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Stop bits:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="stopBox">
           <property name="minimumSize">
            <size>
             <width>125</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="lblFlowControl">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Flow control:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="flowBox">
           <property name="minimumSize">
            <size>
             <width>125</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="2">
          <widget class="QPushButton" name="btnApplyOptions">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Apply currently selected options to port</string>
           </property>
           <property name="text">
            <string>Apply</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <spacer name="verticalSpacerTop">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>5</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="0" colspan="3">
      <widget class="Line" name="line_underline0">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>5</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
